container/snapshot.go

```diff
@@ -0,0 +1,152 @@
+package container
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/go-connections/nat"
+)
+
+// Snapshot is a read only view for Containers
```

Comment:
Please add some comment about how to maintain this structure?
e.g. Why `HostConfig.{NetworkMode, Isolation}` is needed but others not?